riptide-core/src/main/java/org/zalando/riptide/UrlResolution.java

```diff
@@ -0,0 +1,47 @@
+package org.zalando.riptide;
+
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+
+// TODO introduce UrlResolutionStrategy interface in the future, if needed
+public enum UrlResolution {
+
+    /**
+     * Resolves a given relativce URI by following the rules defined in RFC 3986.
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc3986">RFC 3986</a>
+     * @see URI#resolve(URI)
+     */
+    RFC {
+        @Override
+        URI resolve(final URI baseUrl, final URI uri) {
+            return baseUrl.resolve(uri);
+        }
+    },
+
+    /**
+     * Resolves a given relative URI by appending its path to and merging its query parameters with the given base URL.
+     *
+     * @see UriComponentsBuilder#pathSegment(String...)
+     */
+    APPEND {
+        @Override
+        URI resolve(final URI baseUrl, final URI uri) {
+            return UriComponentsBuilder.fromUri(baseUrl)
+                    .pathSegment(uri.getRawPath())
```

Comment:
I tried `path(..)` but it won't work with the following:

Base URL: http://example.org/api
URI: foo

Will produce: **http://example.org/apifoo**